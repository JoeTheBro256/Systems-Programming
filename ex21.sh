#!/bin/bash

# Filename: ex21.sh
#
# Problem: Count the number of quotes each person had in the top 100 quote
# list. Then output the list of names in alphabetical order with the number of quotes each.
cat ex21.input | gawk '{


		if (match($0,"-")) {
			name=substr($0,0,match($0,"-")-1);
			names[name]++;
		}
	}
	END {
			n = asorti(names,namesi)
			#print n
			for (i=1;i<=n;i++){
				print namesi[i] "had", names[namesi[i]],"quotes."
			}
}'
