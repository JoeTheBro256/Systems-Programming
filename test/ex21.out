awk: line 4: syntax error at or near <
awk: line 8: syntax error at or near <
